neptune_tf/sync/tensorflow_data_sync.py

```diff
@@ -0,0 +1,120 @@
+#
+# Copyright (c) 2019, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import print_function
+
+import os
+import random
+import re
+import traceback
+
+import click
+from future.moves import collections, sys
+
+from neptune_tf.integration.tensorflow_integration import TensorflowIntegrator
+
+
+class TensorflowDataSync(object):
+    _RECORD = collections.namedtuple('Record', 'x value')
+
+    def __init__(self, project, path):
+        self._project = project
+        self._path = path
+
+    @staticmethod
+    def requirements_installed():
+        # pylint:disable=unused-variable,unused-import
+        try:
+            import tensorflow
+            return True
+        except ImportError:
+            return False
+
+    def run(self):
+        import tensorflow as tf
+        for root, _, run_files in os.walk(self._path):
+            for run_file in run_files:
+                try:
+                    self._load_single_run(os.path.join(root, run_file), tf)
+                except Exception as e:
+                    print("Cannot load run from file '{}'. ".format(run_file) + str(e), file=sys.stderr)
+                    traceback.print_exc(e)
+
+    def _load_single_run(self, path, tf):
+        click.echo("Loading {}...".format(path))
+        run_path = os.path.relpath(path, self._path)
+        run_name = re.sub(r'[^0-9A-Za-z_\-]', '_', run_path)
+        if not self._experiment_exists(run_name):
+            with self._project.create_experiment(name=run_name,
+                                                 properties={
+                                                     'tf/run/name': run_name,
```

Comment:
Zbędnę. Path wystarczy. Ten name to w sumie jakiś nasz internal do identyfikacji więc może być użyty tylko w tagu, bo tam niestety jest konieczny.